nf/CELLRANGER/main.nf

```diff
@@ -33,24 +48,33 @@ workflow {
     if (params.mode == "count"){
         COUNT(
             sample,
-            create_bam,
             reads,
             mates,
             transcriptome_dir,
             indices
         )
+        sample_analysis_dir = COUNT.out.analysis_dir.map { dirname -> [sample + ".cellranger_analysis", dirname] }
+        TAR_DIR(sample_analysis_dir)
     }
 
     else{
         MULTI(
             library_fastq_id,
-            create_bam,
             reads,
             mates,
             transcriptome_dir,
             feature_types,
             sample_sheet,
             probe_set
         )
+        // extract sample_id from path
+        pattern = /.*\/per_sample_outs\/(\S+)\/count\/analysis/
+        multi_sample_analysis_dir = MULTI.out.multi_analysis.flatten().map{
+            dirname ->
+            def sample_id_match = dirname =~ pattern
```

Comment:
Is there a reason to define the pattern outside the closure? I don't think it makes any difference, just wondering why you did it that way